maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -541,6 +543,18 @@ else if ( cliRequest.commandLine.hasOption( CLIManager.BATCH_MODE )
 
         plexusLoggerManager = new Slf4jLoggerManager();
         slf4jLogger = slf4jLoggerFactory.getLogger( this.getClass().getName() );
+
+        if ( cliRequest.commandLine.hasOption( CLIManager.FAIL_LEVEL ) )
+        {
+            String logLevelThreshold = cliRequest.commandLine.getOptionValue( CLIManager.FAIL_LEVEL );
+
+            if ( slf4jLoggerFactory instanceof MavenSlf4jWrapperFactory )
```

Comment:
I'm missing an `else` here. Is there a situation where this condition is _not_ going to be `true`? If there is, the `fl` switch doesn't work? If so, maybe we should at least say that it's not going to work.

Reply:
Good points, will add some logging on both if-statements!

Reply:
One time should be enough, I hope... ;-)